libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -1169,6 +1169,34 @@ bool AP_Arming::arm_checks(AP_Arming::Method method)
         }
     }
 
+    // ensure the GPS drivers are ready on any final changes
+    if ((checks_to_perform & ARMING_CHECK_ALL) ||
+        (checks_to_perform & ARMING_CHECK_GPS_CONFIG)) {
+        if (!AP::gps().prepare_for_arming()) {
+            return false;
+        }
+    }
+
+    // note that this will prepare AP_Logger to start logging
+    // so should be the last check to be done before arming
+    if (!mandatory_checks(true)) {
+        return false;
+    }
+
+    AP_Logger *logger = AP_Logger::get_singleton();
```

Comment:
Mandatory checks are run before arming checks, which means that this happens before the prepare for arming call, which means we don't check the result of trying to arm as far as I can tell? (IE what if we fail to start logging that we wanted)

Reply:
Right, sorry about that, totally skimmed over what that meant by accident.